src/main/groovy/com/ullink/Msbuild.groovy

```diff
@@ -107,6 +107,18 @@ class Msbuild extends ConventionTask {
                     parseProject = false
                 } else {
                     projectParsed = allProjects[projectName]
+
+                    if (projectParsed == null) {
+                        def potentialProject = allProjects.collect {it.value}.find {
+                            it.properties.containsKey('ProjectDir')
```

Comment:
My first idea would be to detect if there is a *single* project and if not abort the automatic detection. Not sure if it's better to take the first one if there are more or not.